pvl/math/prime_sieve.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef PVL_MATH_PRIME_SIEVE_HPP_
+#define PVL_MATH_PRIME_SIEVE_HPP_ 1
+
+#include <algorithm>
+#include <vector>
+#include <bitset>
+
+namespace pvl {
+
+struct prime_sieve {
+    std::vector<bool> is_composite;
+    std::vector<int> primes;
+    explicit prime_sieve(int n) : is_composite(n, 0) {
+        for (int i = 2; i < n; i++) {
+            if (!is_composite[i]) primes.push_back(i);
+            for (int j = 0;
+                j < primes.size() && i*primes[j] < n;
+                j++) {
```

Comment:
How about this?
```
for (
  int j = 0;
  j < primes.size() && i*primes[j] < n;
  j++
) {
```

Reply:
i just renamed the vector to be shorter. It fits in one line now HAHA